Desktop_Interface/isobuffer.cpp

```diff
@@ -2,391 +2,364 @@
 #include "isodriver.h"
 #include "uartstyledecoder.h"
 
-isoBuffer::isoBuffer(QWidget *parent, int bufferLen, isoDriver *caller, unsigned char channel_value) : QWidget(parent)
+isoBuffer::isoBuffer(QWidget* parent, int bufferLen, isoDriver* caller, unsigned char channel_value)
+	: QWidget(parent)
 {
-    buffer = (short *) calloc(bufferLen*2, sizeof(short));
-    bufferEnd = bufferLen-1;
-    samplesPerSecond = (double) bufferLen/(double)21;
-    samplesPerSecond = samplesPerSecond/375*VALID_DATA_PER_375;
-    sampleRate_bit = samplesPerSecond * 8;
-    virtualParent = caller;
-    channel = channel_value;
+	buffer = (short*) calloc(bufferLen*2, sizeof(short));
+	bufferEnd = bufferLen-1;
+	samplesPerSecond = (double) bufferLen/(double)21;
+	samplesPerSecond = samplesPerSecond/375*VALID_DATA_PER_375;
+	sampleRate_bit = samplesPerSecond * 8;
+	virtualParent = caller;
+	channel = channel_value;
 }
+
 void isoBuffer::openFile(QString newFile)
 {
-    if (fptr != NULL){
-        fclose(fptr);
-    }
-    if (newFile.isEmpty()){
-        fptr = NULL;
-    }
-    else {
-        QByteArray temp = newFile.toLatin1();
-        char *fileName = temp.data();
-        fptr = fopen(fileName, "w");
-        if (fptr == NULL) qFatal("Null fptr in isoBuffer::openFile");
-        qDebug() << "opening file" << fileName;
-        qDebug() << "fptr = " << fptr;
-    }
+	if (fptr != NULL)
+	{
+		fclose(fptr);
+	}
+	if (newFile.isEmpty())
+	{
+		fptr = NULL;
+	}
+	else
+	{
+		QByteArray temp = newFile.toLatin1();
+		char* fileName = temp.data();
+		fptr = fopen(fileName, "w");
+		if (fptr == NULL) qFatal("Null fptr in isoBuffer::openFile");
+		qDebug() << "opening file" << fileName;
+		qDebug() << "fptr = " << fptr;
+	}
+}
+
+void isoBuffer::insertIntoBuffer(short item)
+{
+	buffer[back] = item;
+	back++;
+	insertedCount++;
+	if (insertedCount > bufferEnd)
+	{
+		insertedCount = bufferEnd+1;
+	}
+	if (back > bufferEnd)
```

Comment:
As above: it's best to separate independent if() statements.  Keeping them together can make them look like a single if-else construct at a glance.

Reply:
oops... I didn't notice i did that. I'll sort that out.